Fill.py

```diff
@@ -61,24 +61,35 @@ def distribute_items_restrictive(window, worlds, fill_locations=None):
     prioitempool = [item for item in itempool if not item.advancement and item.priority]
     restitempool = [item for item in itempool if not item.advancement and not item.priority]
 
+    cloakable_locations = shop_locations + song_locations + fill_locations
+    all_models = shopitempool + dungeon_items + songitempool + itempool
+    worlds[0].settings.distribution.fill(window, worlds, [shop_locations, song_locations, fill_locations], [shopitempool, dungeon_items, songitempool, progitempool, prioitempool, restitempool])
+    itempool = progitempool + prioitempool + restitempool
+
     # set ice traps to have the appearance of other random items in the item pool
     ice_traps = [item for item in itempool if item.name == 'Ice Trap']
+    # Extend with ice traps manually placed in plandomizer
+    ice_traps.extend([
+        location.item for location in cloakable_locations
+        if location.name in location_groups['CanSee']
```

Comment:
readability: Can you wrap this if condition in parens and indent the subsequent lines?

Also, for `extend` we can pass a generator, so I'd drop the `[]`s.